Draft: benchmarks: gauge -> tasty-bench

doyougnu opened the pull request:
Hey Levent,

This PR addresses #608. It swaps out `gauge` for `tasty-bench` and removes `SBVBenchmark` target since it is made redundant by `tasty-bench` and the pattern feature of `tasty`. 

I've removed a bunch of now unnecessary code. Other than that this was a drop in API replacement. However, there is just one hangup. It seems that the `standalone` benchmarks (generating a transcript and then call z3 on it directly) are now significantly slower than the `sbv` versions. I suspect this has something to do with how strict I/O is in `gauge`, and might not be as strict in `tasty-bench`. Honestly I have no idea and unfortunately do not really have the bandwidth to continue this right now.

So I suggest removing the `standalone` benchmarks or using a tasty pattern to only run `sbv*` benchmarks. The latter would leave the `standalone` ones in place but just never run them. Right now they take ~10x longer than the sbv benchmarks. 

What are your thoughts? I'm marking this as draft until we've decided on a good way forward.

LeventErkok commented:
@doyougnu 

Thanks Jeff.  I think the current goal is to make sure the benchmark compiles with GHC9.0.2; so I wouldn't worry about actual differences in the timings. I suggest you remove the `standalone` benchmark indeed, and we continue with `sbv*` only. If this becomes and issue in the future, we can address it then. Please remove the `standalone` and turn it into a PR and I'll merge it. Thanks!

doyougnu commented:
@LeventErkok that should be it. Removing `overhead` benchmarks also simplified the benchmark suite quite a bit. I also removed the `silently` dependency, whose sole purpose was suppressing output to `stdout`. Tasty has a `--quiet` flag so I think this is redundant although I didn't test it to be sure. In any case selectively enabling certain benchmarks is probably a better solution due to the maintenance costs of adding dependencies.

All in all by switching from `gauge` to `tasty-bench` we removed 400 lines of code and 7 dependencies. Excellent work @bodigrim

LeventErkok commented:
Fabulous! Thanks..
